### PR TITLE
aws_api_gateway_rest_api: Add execute_arn computed attribute

### PR DESCRIPTION
**Pull request by mewa**
Added execute_arn attribute as described in #3967

### CODE REVIEW

File: aws/resource_aws_api_gateway_rest_api.go [hunk 1]
```diff
@@ -152,6 +156,16 @@ func resourceAwsApiGatewayRestApiRead(d *schema.ResourceData, meta interface{})
 	d.Set("name", api.Name)
 	d.Set("description", api.Description)
 	d.Set("binary_media_types", api.BinaryMediaTypes)
+
+	restApiId := d.Id()
+	region := meta.(*AWSClient).region
+	accountId := meta.(*AWSClient).accountid
+	arn, err := buildApiGatewayExecutionARN(restApiId, region, accountId)
```
Comment (bflad) [inline aws/resource_aws_api_gateway_rest_api.go]: This function no longer exists as we have moved to prefer using the SDK `arn` package. Can you please build the ARN like the following (modifying as appropriate):

```go
executionArn := arn.ARN{
		Partition: meta.(*AWSClient).partition,
		Service:   "execute-api",
		Region:    meta.(*AWSClient).region,
		AccountID: meta.(*AWSClient).accountid,
		Resource:  fmt.Sprintf("%s/%s", restApiId, stageName),
	}.String()
	d.Set("execution_arn", executionArn)
```

File: website/docs/r/api_gateway_rest_api.html.markdown [hunk 2]
```diff
@@ -12,13 +12,30 @@ Provides an API Gateway REST API.
...
+### Defining REST API
+
 ```hcl
 resource "aws_api_gateway_rest_api" "MyDemoAPI" {
   name        = "MyDemoAPI"
   description = "This is my API for demonstration purposes"
 }
 ```
 
+### Using REST API execution_arn to specify Lambda permissions
```
Comment (bflad) [inline website/docs/r/api_gateway_rest_api.html.markdown]: Can you please move this example to the `aws_lambda_permission` resource documentation instead?

**Comment by mewa** [comment]
Hi @bflad, thank you for your response (and sorry for replying late). I have applied your suggestions and merged changes from master, so hopefully the PR should be good to go.

**Comment by bflad** [comment]
This has been released in [version 1.19.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1190-may-16-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**